build_jaxlib/build.sh

```diff
@@ -0,0 +1,287 @@
+if [[ $? -ne 0 ]]; then
+    exit $1
+fi
+
+eval set -- "$args"
+while [ : ]; do
+    case "$1" in
+        --build-param)
+            BUILD_PARAM="$BUILD_PARAM $2"
+            shift 2
+            ;;
+        -h | --help)
+            usage 1
+            ;;
+        --clean)
+            CLEAN=1
+            shift 1
+            ;;
+        --clean-only)
+            CLEANONLY=1
+            shift 1
+            ;;
+        --cpu-arch)
+            CPU_ARCH="$2"
+            shift 2
+            ;;
+        --debug)
+            DEBUG=1
+            shift 1
+            ;;
+        --dry)
+            DRY=1
+            shift 1
+            ;;
+        --editable)
+            EDITABLE=1
+            shift 1
+            ;;
+        --nccl)
+            NCCL="$2"
+            if [[ $NCCL != "local" ]] && [[ $NCCL != "download" ]]; then
+                echo "Unsupported argument to --nccl"
+                usage 1
+            fi
+            shift 2
+            ;;
+        --src-path-jax)
+            SRC_PATH_JAX=$2
+            shift 2
+            ;;
+        --src-path-xla)
+            SRC_PATH_XLA=$2
+            shift 2
+            ;;
+        --sm)
+            CUDA_COMPUTE_CAPABILITIES=$2
+            shift 2
+            ;;
+        --)
+            shift;
+            break 
+            ;;
+        *)
+            echo "UNKNOWN OPTION $1"
+            usage 1
+    esac
+done
+
+## Set internal variables
+
+SRC_PATH_JAX=$(realpath $SRC_PATH_JAX)
+SRC_PATH_XLA=$(realpath $SRC_PATH_XLA)
+
+export DEBIAN_FRONTEND=noninteractive
+export TZ=America/Los_Angeles
+
+export TF_NEED_CUDA=1
+export TF_NEED_CUTENSOR=1
+export TF_NEED_TENSORRT=0
+export TF_CUDA_PATHS=/usr,/usr/local/cuda
+export TF_CUDNN_PATHS=/usr/lib/$(uname -p)-linux-gnu
+export TF_CUDA_VERSION=$(ls /usr/local/cuda/lib64/libcudart.so.*.*.* | cut -d . -f 3-4)
+export TF_CUBLAS_VERSION=$(ls /usr/local/cuda/lib64/libcublas.so.*.*.* | cut -d . -f 3)
+export TF_CUDNN_VERSION=$(echo "${NV_CUDNN_VERSION}" | cut -d . -f 1)
+
+case "${NCCL}" in
+    "local")
+        export TF_NCCL_VERSION=$(echo "${NCCL_VERSION}" | cut -d . -f 1)
+        ;;
+    "download")
+        unset TF_NCCL_VERSION
+        ;;
+esac
+
+case "${CPU_ARCH}" in
+    "amd64")
+        export CC_OPT_FLAGS="-march=sandybridge -mtune=broadwell"
+        ;;
+    "arm64")
+        export CC_OPT_FLAGS="-march=armv8-a"
+        ;;
+esac
+
+if [[ ! -z "${CUDA_COMPUTE_CAPABILITIES}" ]]; then
+    if [[ "$CUDA_COMPUTE_CAPABILITIES" == "all" ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=$(supported_compute_capabilities ${CPU_ARCH})
+        if [[ $? -ne 0 ]]; then exit 1; fi
+    elif [[ "$CUDA_COMPUTE_CAPABILITIES" == "local" ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=$(./local_cuda_arch)
+    else
+        export TF_CUDA_COMPUTE_CAPABILITIES="${CUDA_COMPUTE_CAPABILITIES}"
+    fi
+fi
+
+if [[ "${EDITABLE}" == "1" ]]; then
+    BUILD_PARAM="${BUILD_PARAM} --editable"
+fi
```

Comment:
why don't we do this directly in the flag parsing while loop above?
you then don't need the additional EDITABLE variable.